python/ray/train/lightning/lightning_checkpoint.py

```diff
@@ -0,0 +1,60 @@
+from typing import TYPE_CHECKING, Optional, Type, Union, Dict
+
+from ray.air.checkpoint import Checkpoint
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    from pytorch_lightning import LightningModule
+    from ray.data.preprocessor import Preprocessor
+
+
+@PublicAPI(stability="alpha")
+class LightningCheckpoint(Checkpoint):
+    """A ``ray.air.checkpoint.Checkpoint`` with Lightning-specific functionality.
+
+    Create this from a generic :py:class:`~ray.air.checkpoint.Checkpoint` by calling
+    ``LightningCheckpoint.from_checkpoint(ckpt)``.
+    """
+
+    @classmethod
+    def from_model(
+        cls,
+        model: "LightningModule",
+        model_params: Dict,
+        *,
+        preprocessor: Optional["Preprocessor"] = None,
+    ) -> "LightningCheckpoint":
+        """Create a :py:class:`~ray.air.checkpoint.Checkpoint` that stores a Lightning
+        module.
+
+        Args:
+            model: The LightningModule to store in the checkpoint.
+            model_params: The kwargs to be passed to ``model``'s initialization.
+            preprocessor: A fitted preprocessor to be applied before inference.
+
+        Returns:
+            An :py:class:`LightningCheckpoint` containing the specified model.
+
+        Examples:
+            >>> from ray.train.lightning import LightningCheckpoint
+            >>> from pytorch_lightning import LightningModule
+            >>>
+            >>> model = LightningModule()
+            >>> checkpoint = LightningCheckpoint.from_model(model)
+        """
+        pass
+
+    def get_model(
+        self, model: Union[Type["LightningModule"], None]
+    ) -> "LightningModule":
```

Comment:
I think the reason I didn't make the `model` arg a kwarg that defaults to None was because I thought it made sense to require users to be more intentional about usage

BUT I think that since we can raise an actionable error if they incorrectly use this arg (see below), it's probably okay to make this default to None and users can update if told to by the error
https://github.com/ray-project/ray/blob/6358cde13d8a03ebb894ab4c0ccb58f467da7aae/python/ray/train/lightning/_lightning_utils.py#L120-L125